pkg/api/v20230701preview/openshiftcluster.go

```diff
@@ -133,6 +133,41 @@ const (
 	OutboundTypeLoadbalancer       OutboundType = "Loadbalancer"
 )
 
+// ResourceReference represents a reference to an Azure resource.
+type ResourceReference struct {
+	// The fully qualified Azure resource id.
+	ID string `json:"id,omitempty"`
+}
+
+// LoadBalancerProfile represents the profile of the cluster public loadbalancer.
```

Comment:
nit: "load balancer"

Reply:
I agree that condition is not the greatest, How do you feel about updating that condition to `np.LoadbalancerProfile != nil && np.OutboundType != OutboundTypeLoadbalancer && np.OutboundType != ""`, assuming the default outboundType will always be Loadbalancer we should be OK.  IF default outboundType was changed, then I think this would break without a unit test failing.

I will fix the nit.